addons/udes_priorities/models/priorities.py

```diff
@@ -0,0 +1,59 @@
+# -*- coding: utf-8 -*-
+
+from odoo import models, fields, api, _
+
+
+class UdesPriority(models.Model):
+    _name = "udes_priorities.priority"
+    _order = "sequence,id"
+
+    name = fields.Char(required=True)
+    reference = fields.Char(help="Unique identifier for the priority", required=True)
+
+    def _default_sequence_id(self):
+        return self.env.ref("udes_priorities.sequence_default")
+
+    sequence = fields.Integer(default=_default_sequence_id, help="Used to order the priorities")
+    description = fields.Text(help="A description about the perpose of the priority")
+    picking_type_ids = fields.Many2many(
+        "stock.picking.type",
+        string="Stock Transfer Types",
+        help="The stock picking types this priority is visible on. No picking types means all picking types",
+    )
+    group_ids = fields.Many2many(
+        "udes_priorities.priority_group", column1="priority_id", column2="group_id"
+    )
+
+    active = fields.Boolean(default=True)
+
+    _sql_constraints = [
+        ("priority_reference_unique", "unique(reference)", "reference must be unique")
+    ]
+
+    @api.multi
+    def _number_of_oustanding_pickings(self):
+        Picking = self.env["stock.picking"]
+        self.ensure_one()
+        return Picking.search_count(
+            [("priority", "=", self.reference), ("state", "not in", ("cancel", "done")),]
```

Comment:
If we drop the `ensure_one` and change the first search clause to `("priority", "in", self.mapped("reference"))`, then we can have `_check_for_oustanding_pickings` do a single query instead of one per priority.

Reply:
I updated this so it informs us which priorities have the outstanding pickings which probably better UX so the user knows which one is breaking. However, that does require one query per priority or returning the picking them selves.